api/src/controllers/extensions.ts

```diff
@@ -47,7 +47,7 @@ router.get(
 		res.setHeader('Content-Type', 'application/javascript; charset=UTF-8');
 		res.setHeader(
 			'Cache-Control',
-			env.EXTENSIONS_CACHE_TTL ? getCacheControlHeader(req, ms(env.EXTENSIONS_CACHE_TTL as string)) : 'no-store'
+			env.EXTENSIONS_CACHE_TTL ? `max-age=${ms(env.EXTENSIONS_CACHE_TTL as string) / 1000}` : 'no-store'
```

Comment:
Makes sense! But one thing that sparked my curiosity is the following rounding in `getCacheControlHeader()`:

https://github.com/directus/directus/blob/e96e04efa09394c495d519c766c6deb02ce0a195/api/src/utils/get-cache-headers.ts#L23-L24

Although the referred `ASSETS_CACHE_TTL` code doesn't round, I wonder should both `ASSETS_CACHE_TTL` and `EXTENSIONS_CACHE_TTL` be ensured to be rounded instead?

With a quick test of setting `ASSETS_CACHE_TTL` to `1234ms`, I was able to get `max-age=1.234` which I believe is not an intended behavior 🤔 Example screenshot:

![](https://user-images.githubusercontent.com/42867097/217808904-96c9d06c-edb0-41d6-8cbf-c6fcc56d6341.png)

---

Interestingly I was also wondering whether this issue is present for `REFRESH_TOKEN_TTL` usage here as we aren't rounding it:

https://github.com/directus/directus/blob/e96e04efa09394c495d519c766c6deb02ce0a195/api/src/constants.ts#L58

but seems like express `res.cookie` does use `Math.floor(val / 1000)` instead of `Math.round` as seen here: https://github.com/expressjs/express/blob/442fd467992992558806da8da07e945838712587/lib/response.js#L876

so in the case of `REFRESH_TOKEN_TTL` being used in `maxAge` for cookie options, we can leave it as milliseconds as-is.

Reply:
Another question is do we need to keep the `private`/`public` in front of max-age that was present in `getCacheControlHeader()` and the referred `ASSETS_CACHE_TTL` code? 

I'm just bringing this up in case it was lost during the change, but if you were intentionally removing it (since it's public), then feel free to dismiss this 👍

Reply:
> Although the referred ASSETS_CACHE_TTL code doesn't round, I wonder should both ASSETS_CACHE_TTL and EXTENSIONS_CACHE_TTL be ensured to be rounded instead?

Very good catch! Indeed, I think we should round the value in both cases. I'm going to adjust it accordingly 👍 

> I'm just bringing this up in case it was lost during the change, but if you were intentionally removing it (since it's public), then feel free to dismiss this 👍

Yes, I've omitted it intentionally since it's public 👍

Reply:
Now using `getCacheControlHeader()` function for both cases, where care is taken to round the value.